lib/puppet/pal/pal_impl.rb

```diff
@@ -434,6 +442,22 @@ def self.main(manifest, facts, variables, internal_compiler_class)
           overrides[:pal_catalog_compiler] = overrides[:pal_compiler] = pal_compiler
         end
 
+        # When scripting the trusted data are always local, but set them anyway
+        # When compiling for a catalog, the catalog compiler does this
+        unless internal_compiler_class == :catalog || disable_facts
```

Comment:
`with_catalog_compiler` should pass the appropriate value when calling `main` rather than having this check hardcoded here.

Reply:
Wait it _does_ pass the appropriate value. So the `internal_compiler_class` check should be unnecessary?